query/physicalplan/filter.go

```diff
@@ -91,16 +93,38 @@ func binaryBooleanExpr(expr *logicalplan.BinaryExpr) (BooleanExpression, error)
 
 		switch expr.Op {
 		case logicalplan.OpRegexMatch:
-			regexp, err := regexp.Compile(string(rightScalar.(*scalar.String).Data()))
+			rightString := string(rightScalar.(*scalar.String).Data())
+
+			// This is a simple optimization to avoid using regex for simple substring matches.
+			if strings.HasPrefix(rightString, ".*") && strings.HasSuffix(rightString, ".*") {
```

Comment:
This is not correct, there can be other wildcards in the regex and then this doesn't work.